### PR TITLE
Order details page as guest displays empty order (GH-6480)

### PR DESCRIPTION
**Pull request by qjimmy**
Closes GH-6480 for 1.5.x branch

Changed redirect to login page instead of custom guard for order details page

### CODE REVIEW

**Review by bgambocjaviniar — COMMENTED** [review]
As I can't be the one to review it as it's under the code owners thing, I would like to leave a comment saying

do we really need to re-create the guard? Reading the comments, I see they mentioned to put it to homepage and that is why this came to be, but I believe as any protected pages that shouldn't be accessed by people unless `authenticated`, they should be redirected to `login page` as we want them to register. I believe we need to stay consistent to what we have in the application and not redirect them to homepage.

However, if we really need to redirect them to homepage, I think we need to adapt the guard and make it configurable, but I think the first point on just protected the page and redirecting them to login is enough.

I believe you need to follow up with someone on this.

**Comment by qjimmy** [comment]
Just talked to Gil for a second opinion, login page seems like the right page to redirect to

**Review by hackergil — APPROVED** [review]

**Pull request merged**
